### ISSUE TITLE
Replay format tests use `println!` instead of `assert!`ing things

### ISSUE THREAD

**Issue by ralokt** [eee555/ms-toollib#6]
I took a look at the replay format tests, and they mostly or exclusively write information about the replays they test to standard output.

While that is of course still useful, as the replay files need to be successfully parsed in order for the tests to pass, checking the contents would provide more comprehensive coverage and be able to catch subtle bugs where output is subtly wrong for some reason.

That of course requires the replay files to be consistent - but if #5 is implemented, that is guaranteed.

**Comment by ralokt** [comment]
#7 contains an example of what I mean :)

**Comment by eee555** [comment]
I understand what you mean. Using println! may not be strict enough, but it allows me to develop faster.

**Comment by ralokt** [comment]
I do see how it can be nice to just see everything that's going on without stopping at the first wrong assertion. Both is good? :)

**Comment by eee555** [comment]
Every time a file with a parsing exception is encountered, the file needs to be replaced and tested again. If using "assert", many changes will be required, whereas "print" doesn't need modifications. These details are up to whoever is working on it to decide. Now since that codes are relatively stable, replacing print with assert would also be fine.

**Comment by ralokt** [comment]
Ah, I see!

I'd recommend the following workflow here:
 - have an executable script that does what the test currently does
 - run the script on replays that cause problems to debug
 - use the output of the script to easily generate a new test for every broken replay that comes in

That way, the tests are clean and precise, but you have a way to quickly debug things. In theory, if the tests are all the same, just with different values, there could even be a macro 🤔

**Comment by eee555** [comment]
It is best to avoid uploading binary files, as the principle is that only text files should be uploaded to Git.

**Comment by ralokt** [comment]
> It is best to avoid uploading binary files, as the principle is that only text files should be uploaded to Git.

I disagree with that version of the principle, although I see where it might come from, because I strongly agree with a subset of it - compilation results.

Having those versioned is obviously bad - it's redundant, the repository contains the code that the binary is compiled from, and if a developer forgets to commit changes to one or the other, there's a lot of potential for confusion...

(Side note: I even kinda dislike lockfiles for that reason, even though there are good arguments for versioning those. I've done it, but it took me years to warm up to the concept as even though they are text, they are technically compilation outputs for your dependency declarations.)

However, this is different. These aren't compilation outputs, these are test assets that happen to be binary. Yes, delta compression doesn't work on them, and this becomes an issue if you were to, for example, version a bunch of images that get edited - but also, replay files won't change.

The way I'd rephrase the principle is: Anything that isn't necessary to reproducibly build, run, and test the project from scratch shouldn't be versioned in git (and also, be careful with binary file sizes).

It's your call ofc - just offering this as a less limiting and more nuanced approach :)

**Comment by ralokt** [comment]
Awesome! Sorry for the late reply - my life is super busy right now, but I hope to get back to contributing here soon. But this change is great to see! :)
